client/src/components/geoJSONMap.tsx

```diff
@@ -30,16 +30,17 @@ const GeoJSONMap: React.FC<GeoJSONMapProps> = ({ geoJsonData }) => {
     }>({});
     const [allValues, setValues] = useState<number[]>([]);
     const [steps, setSteps] = useState<number>(5); // State for steps
-    const { colorPickerColor, featureVisibility } = useToggle();
+    const {primaryColorPicker, secondaryColorPicker, featureVisibility, autoColourRange} = useToggle();
 
     // Effect to initialize color gradient and data values
     useEffect(() => {
         if (geoJsonData) {
             setValues(extractValuesFromGeoJSON(geoJsonData));
-            const rgbColor = hexToRgb(colorPickerColor);
-            setColorGradient(generateColorGradient(steps, rgbColor));
+            const rgbColor = hexToRgb(primaryColorPicker);
+            const rgbColor_2 = hexToRgb(secondaryColorPicker);

```

Comment:
the var names could be more verbose.

Reply:
So fair, think I made these just as placeholders to test then forgot to change them.